app/transformers/PAYETransformer.scala

```diff
@@ -0,0 +1,300 @@
+            middleTierAmountJson("total_income_before_tax", incomeBeforeTax.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_data \ 'payload,
+            middleTierAmountJson("benefits_from_employment", employmentBenefits.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_data \ 'payload,
+            middleTierAmountJson("taxable_state_benefits", taxableStateBenefits.getOrElse(0)))
+
+      safeTransform(jsonTransformer)
+    }
+    def transformSummary(source: JsObject): JsObject = {
+      val taxableIncome: Option[Double] = pickAmount(__ \ 'income \ 'incomeBeforeTax, source)
+      val taxFreeAmount: Option[Double] = pickAmount(__ \ 'income \ 'taxableIncome, source)
+      val incomeTaxAndNics: Option[Double] = pickAmount(__ \ 'calculatedTotals \ 'totalIncomeTaxNics, source)
+      val IncomeAfterTaxAndNics: Option[Double] = pickAmount(__ \ 'calculatedTotals \ 'incomeAfterTaxNics, source)
+      val incomeTax: Option[Double] = pickAmount(__ \ 'calculatedTotals \ 'totalIncomeTax2, source)
+      val nationalInsurance: Option[Double] = pickAmount(__ \ 'nationalInsurance \ 'employeeContributions, source)
+      val nationalInsuranceEmployer: Option[Double] =
+        pickAmount(__ \ 'nationalInsurance \ 'employerContributions, source)
+      val averageIncomeTaxRate: Option[Double] = pickAmount(__ \ 'averageRateTax, source)
+
+      val jsonTransformer =
+        appendAttribute(
+          __ \ 'summary_data \ 'payload,
+          middleTierAmountJson("total_income_before_tax", taxableIncome.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'payload,
+            middleTierAmountJson("total_tax_free_amount", taxFreeAmount.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'payload,
+            middleTierAmountJson("total_income_tax_and_nics", incomeTaxAndNics.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'payload,
+            middleTierAmountJson("income_after_tax_and_nics", IncomeAfterTaxAndNics.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'payload,
+            middleTierAmountJson("total_income_tax", incomeTax.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'payload,
+            middleTierAmountJson("employee_nic_amount", nationalInsurance.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'payload,
+            middleTierAmountJson("employer_nic_amount", nationalInsuranceEmployer.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'payload,
+            middleTierAmountJson("nics_and_tax_rate_amount", averageIncomeTaxRate.getOrElse(0), "PERCENT")) andThen
+          appendAttribute(
+            __ \ 'summary_data \ 'rates,
+            middleTierRateJson("nics_and_tax_rate", averageIncomeTaxRate.getOrElse(0)))
```

Comment:
This part of the model doesn't seem to make any sense. Why is the same value in the rates and amounts? It's either a rate, or an amount. It can't be both.